open_rarity/rarity_ranker.py

```diff
@@ -53,18 +54,37 @@ def rank_collection(
         assert len(tokens) == len(scores)
 
         token_rarities: list[TokenRarity] = []
+
         # augment collection tokens with score information
         for idx, token in enumerate(tokens):
-            token_rarities.append(TokenRarity(token=token, score=scores[idx]))
+
+            # extract features from the token
+            token_features = (
+                TokenFeatureExtractor.extract_unique_attribute_count(
+                    token=token, collection=collection
+                )
+            )
+
+            token_rarities.append(
+                TokenRarity(
+                    token=token,
+                    score=scores[idx],
+                    token_features=token_features,
+                )
+            )
 
         return RarityRanker.set_rarity_ranks(token_rarities)
 
     @staticmethod
     def set_rarity_ranks(
         token_rarities: list[TokenRarity],
     ) -> list[TokenRarity]:
-        """Ranks a set of tokens Scores that are higher indicate a higher rarity,
-        and thus a lower rank.
+        """Ranks a set of tokens according to OpenRarity algorithm.
+        To account for unique items in collection OpenRarity implements
+        two-stage sorting ranking based on two factors:
+        Information Content score and unique attributes count. The sorting
+        function takes unique attributes count as a primary factor and
+        the Information Content score as a secondary factor.
```

Comment:
nit: Can simplify language to be:

```
Ranks a set of tokens according to OpenRarity algorithm.
To account for additional factors like unique items in a collection, 
OpenRarity implements multi-column sort. Current sort algorithm uses two factors:
unique attributes count and Information Content score, in order.
```

Reply:
Ranks a set of tokens according to OpenRarity algorithm.
To account for additional factors like unique items in a collection, 
OpenRarity implements `multi-factor sort`. Current sort algorithm uses two factors:
unique attributes count and Information Content score, in order.